apps/bttester/src/btp_gatt.c

```diff
@@ -1871,44 +1871,25 @@ notify_multiple(uint16_t conn_handle, void *arg)
 }
 
 static uint8_t
-set_mult(const void *cmd, uint16_t cmd_len,
-         void *rsp, uint16_t *rsp_len)
+notify_mult(const void *cmd, uint16_t cmd_len,
+            void *rsp, uint16_t *rsp_len)
 {
-    const struct btp_gatt_set_mult_val_cmd *cp = cmd;
-    struct ble_gatt_notif tuples[16];
-    int i;
-    int rc = 0;
-    int data_idx = 0;
-    uint16_t data_len;
+    const struct btp_gatt_notify_mult_val_cmd *cp = cmd;
     struct notify_mult_cb_data cb_data;
+    int i;
 
-    for (i = 0; i < cp->count; i++) {
-        tuples[i].handle = get_le16(cp->data + data_idx);
-        data_idx += 2;
-        tuples[i].value = ble_hs_mbuf_att_pkt();
-        if (tuples[i].value == NULL) {
-            rc = ENOMEM;
-            goto done;
-        }
-
-        data_len = get_le16(cp->data + data_idx);
-        data_idx += 2;
-
-        os_mbuf_append(tuples[i].value, cp->data + data_idx, data_len);
-        data_idx += data_len;
+    if (cp->count > sizeof(cb_data.handles)) {
```

Comment:
while you are at is please also add size validations for btp command since it is declared BTP_HANDLER_LENGTH_VARIABLE

if (cmd_len < sizeof(*cp) || cmd_len != (sizeof(*cp) + (le16toh(cp->count)) * sizeof(cp->handles[0])))) { return failed;}

Reply:
Done